satpy/multiscene.py

```diff
@@ -45,14 +45,45 @@
 log = logging.getLogger(__name__)
 
 
-def stack(datasets):
-    """Overlay series of datasets on top of each other."""
+def stack(datasets, weights=None):
+    """Combine a series of datasets together.
+
+    On default, datasets are stacked on top of each other, so the last one is
+    on top. But if a sequence of weights arrays are provided the datasets will
+    be combined according to those weights. The result will be a composite
+    dataset where the data in each pixel is coming from the dataset having the
+    highest weight.
+
+    """
+    if weights:
+        return _stack_weighted(datasets, weights)
+
     base = datasets[0].copy()
     for dataset in datasets[1:]:
-        base = base.where(dataset.isnull(), dataset)
+        try:
+            base = base.where(dataset == dataset._FillValue, dataset)
+        except AttributeError:
+            base = base.where(dataset.isnull(), dataset)
```

Comment:
Hmm, I thought it should be fine. You mean the way we do here with a try-except, rather than more carefully checking/analysing what kind of data we are working with?

Reply:
Sorry, I should have been more clear. I meant checking the `_FillValue`. I'd be afraid that there is some reader somewhere producing floating point data with a `_FillValue` where that `_FillValue` came from the file attributes and is no longer valid (after the data was unscaled with `scale_factor` and `add_offset`). Probably not worth it too worry about it now but instead fix it when we find the issue.